src/dbnode/storage/bootstrap/bootstrapper/persist.go

```diff
@@ -253,6 +261,14 @@ func BuildBootstrapIndexSegment(
 	retentionOpts := ns.Options().RetentionOptions()
 	nowFn := resultOpts.ClockOptions().NowFn()
 	earliestRetentionTime := retention.FlushTimeStart(retentionOpts, nowFn())
+
+	// If bootstrapping is taking more time than our retention period, we might end up in a situation
+	// when earliestRetentionTime is larger than out block end time. This means that the blocks
+	// got outdated during bootstrap so we just skip building index segments for them.
+	if earliestRetentionTime.After(blockEnd) {
```

Comment:
Shouldn't this also be true when `earliestRetentionTime` equals `blockEnd`?

Reply:
Well, this is the question about block's upper range inclusiveness. If we have a block from `06:00` to `10:00`, do we treat it like `[06:00,10:00]` or `[06:00, 10:00)`? I wasn't sure, so to me it was safer to allow indexing when `earlierRetentionTime` is equals `blockEnd` then to forbid it.